hoomd/md/external/wall.py

```diff
@@ -0,0 +1,653 @@
+          thus is not periodic. Forces will be evaluated and added to all
+          particles from all walls in the wall group. Additionally there are no
+          safeguards requiring a wall to exist inside the box to have
+          interactions. This means that an attractive force existing outside the
+          simulation box would pull particles across the periodic boundary where
+          they would immediately cease to have any interaction with that wall.
+          It is therefore up to the user to use walls in a physically meaningful
+          manner. This includes the geometry of the walls, their interactions,
+          and as noted here their location.
+        - When :math:`r_{\mathrm{cut}} \le 0` or is set to False the particle
+          type wall interaction is excluded.
+        - While wall potentials are based on the same potential energy
+          calculations as pair potentials, Features of pair potentials such as
+          specified neighborlists, and alternative force shifting modes are not
+          supported.
+    """
+
+    def __init__(self, walls):
+        self._walls = None
+        if walls is None:
+            walls = []
+        self.walls = hoomd.wall._WallsMetaList(walls, _to_md_cpp_wall)
+
+    def _attach(self):
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            cls = getattr(_md, self._cpp_class_name)
+        else:
+            cls = getattr(_md, self._cpp_class_name + "GPU")
+        self._cpp_obj = cls(self._simulation.state._cpp_sys_def)
+        self._walls._sync({
+            hoomd.wall.Sphere:
+                _ArrayViewWrapper(
+                    _WallArrayViewFactory(self._cpp_obj, hoomd.wall.Sphere)),
+            hoomd.wall.Cylinder:
+                _ArrayViewWrapper(
+                    _WallArrayViewFactory(self._cpp_obj, hoomd.wall.Cylinder)),
+            hoomd.wall.Plane:
+                _ArrayViewWrapper(
+                    _WallArrayViewFactory(self._cpp_obj, hoomd.wall.Plane)),
+        })
+        super()._attach()
+
+    @property
+    def walls(self):
+        """`list`[ `hoomd.wall.WallGeometry`]: \
+            The walls associated with this wall potential."""
+        return self._walls
+
+    @walls.setter
+    def walls(self, wall_list):
+        if self._walls is wall_list:
+            return
+        self._walls = hoomd.wall._WallsMetaList(wall_list, _to_md_cpp_wall)
+
+
+class LJ(WallPotential):
+    r"""Lennard-Jones wall potential.
+
+    Args:
+        walls (`list` [`hoomd.wall.WallGeometry` ]): A list of wall definitions
+            to use for the potential.
+
+    Wall force evaluated using the Lennard-Jones potential.  See
+    `hoomd.md.pair.LJ` for force details and base parameters and `WallPotential`
+    for generalized wall potential implementation.
+
+    Example::
+
+        walls = [hoomd.wall.Sphere(radius=4.0)]
+        lj = hoomd.md.wall.LJ(walls)
```

Comment:
```suggestion
        lj = hoomd.md.external.wall.LJ(walls=walls)
```
Fix `external` module use and encourage users to use keyword arguments.